### PR TITLE
Implement suitable fallback for Nav block on front end of site when no menu selected

### PR DESCRIPTION
**Pull request by getdave**
## Description

In https://github.com/WordPress/gutenberg/issues/36721 we learnt that the Nav block doesn't currently have a suitable fallback on the front end of the site when a menu is not selected.

This PR implements in the simplest way possible by falling back to rendering a `core/page-list` block with it's default attributes.

As Nav is already set up to handle displaying this block the styles all work perfectly out of the box.

~if we want to limit the number of items shown then we'll need to modify `core/page-list` to have a `maxPages` prop so we can control the number of items. We might be able to provide this via context. I suggest we do this in a separate PR.~ **Update**: I added a commit which introduces a `maxPages` attribute to the Nav block. We can then pass this to the Page List block via context to control the number of items that are displayed.  

Addresses _part of_ https://github.com/WordPress/gutenberg/issues/36721

## How has this been tested?

Before testing use Fakerpress Plugin (or similar) to add a load of pages to your site. Some should be nested parent/child.

1. Open Site Editor.
2. Remove any existing Nav block.
3. Add Nav block from scratch. Don't add any items!
4. Save your empty Nav block and publish the post.
5. Go to front of site.
6. See Page List rendered by default even though no Menu has been selected.
7. Return to Editor and add a menu for your nav block.
8. Save and return to front of site.
9. Check that your menu is rendered.

## Screenshots 

Screenshot 2021-11-22 at 13 16 37

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/navigation/index.php [hunk 1]
```diff
@@ -183,7 +185,7 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 		$inner_blocks            = new WP_Block_List( $parsed_blocks, $attributes );
 	}
 
-	if ( ! empty( $block->context['navigationArea'] ) ) {
+	if ( false && ! empty( $block->context['navigationArea'] ) ) {
```
Comment (getdave) [inline packages/block-library/src/navigation/index.php]: I need to revert this change.

File: packages/block-library/src/navigation/index.php [hunk 2]
```diff
@@ -214,8 +216,17 @@ function( $block ) {
 		$inner_blocks = new WP_Block_List( $compacted_blocks, $attributes );
 	}
 
+	// If there are no inner blocks then fallback to rendering the Page List block.
 	if ( empty( $inner_blocks ) ) {
-		return '';
+		$is_fallback            = true; // indicate we are rendering the fallback.
+		$attributes['maxPages'] = 4; // set value to be passed as context to Page List block.
```
Comment (adamziel) [inline packages/block-library/src/navigation/index.php]: Can we drop maxPages as an attribute from this PR and just use the number 4 directly? I'd rather discuss that separately, like what kind of UI is needed to manage that in the editor.

**Comment by getdave** [comment]
Some additional feedback here https://github.com/WordPress/gutenberg/issues/36721#issuecomment-975585150

File: packages/block-library/src/navigation/index.php (hunk 2 above)
Comment (getdave) [reply packages/block-library/src/navigation/index.php]: I don't think we can easily do that can we? The Page List should only limit the number of items shown when:

- it is inside a Navigation block
- the Nav block has detected no menu is set. 

If we hardcode 4 into the Page List then it will always render only 4 items. That's why I decided to use context.

Open to suggestions.

Comment (getdave) [reply packages/block-library/src/navigation/index.php]: We could add an attribute directly to Page List itself. The block seems to be different though in that it gets all it's properties from context and has no attributes of its own. 

This seems like an intentional design decision so I'd be loath to override that without speaking to someone who worked on that. Perhaps @tellthemachines?

Also something to bear in mind is that one day we might want to limit the number of items shown in the Nav block itself as well. For example, the scenario where you have the "4 corners" navigation - as the theme author you would want to force limit the Nav block to only _visually_ display 4 items (even if it has more data available).

Comment (adamziel) [reply packages/block-library/src/navigation/index.php]: Ah you are right, only the attributes are passed down via a context. Hmmm... I guess there isn't much choice then, is there?

Comment (getdave) [reply packages/block-library/src/navigation/index.php]: > I guess there isn't much choice then, is there?

That's where I got to. It's either use context (as I'm doing here) or we add an attribute directly to the `core/page-list` block and set it from the Nav block.

Comment (adamziel) [reply packages/block-library/src/navigation/index.php]: I wonder if the Query Loop block would provide maxPages "for free". Maybe it could even remove the need for a separate "page-list" block? Let's move on with an unstable attribute and follow up with query loop explorations.

Comment (getdave) [reply packages/block-library/src/navigation/index.php]: If we really don't like `maxPages` let's:

- revert `maxPages`.
- merge this PR
- raise a follow up _with_ `maxPages` for consideration.

**Review by adamziel — APPROVED** [review]

File: packages/block-library/src/navigation/index.php (hunk 2 above)
Comment (adamziel) [reply packages/block-library/src/navigation/index.php]: `__unstable` does the trick and makes it easy to change the implementation if needed. `maxPages` has to be at least a `page-list` attribute, there is no getting rid of it easily.

**Comment by getdave** [comment]
As a followup I'm going to implement finding a Nav Menu. I have the code ready once this is merged.

**Update**: it's ready at https://github.com/WordPress/gutenberg/pull/36740
